ktor-http/common/src/io/ktor/http/ContentTypes.kt

```diff
@@ -279,7 +279,19 @@ public class BadContentTypeFormatException(value: String) : Exception("Bad Conte
 /**
  * Creates a copy of `this` type with the added charset parameter with [charset] value.
  */
-public fun ContentType.withCharset(charset: Charset): ContentType = withParameter("charset", charset.name)
+public fun ContentType.withCharset(charset: Charset): ContentType =
+    withParameter("charset", charset.name)
+
+/**
+ * Creates a copy of `this` type with the added charset parameter with [charset] value
+ * if charset is not ignored by this [ContentType]
+ */
+public fun ContentType.withCharsetIfNeeded(charset: Charset): ContentType =
+    if (charset.name == "UTF-8" && contentType == "application" && contentSubtype == "json") {
```

Comment:
I don't know if it's correct to single out UTF-8 here.

The `application/json` content type does not have _any_ charset parameter, regardless of value. [RFC 4627](https://datatracker.ietf.org/doc/html/rfc4627#section-3) specifies that the character encoding of a JSON text is always auto-detected by the client (from among a limited set of options).

So `application/json; charset=UTF-16` would incorrect just as `application/json; charset=UTF-8` would be. Neither is a valid content type.

Reply:
You are right, thanks for the help! 
I've ignored only UTF-8 because I thought Ktor ignored only this charset with application/json in 1.x.x versions. But I've checked the 1.6.7 version now and I see I was wrong.